metaflow/plugins/kfp/kfp.py

```diff
@@ -336,6 +392,21 @@ def build_kfp_dag(node: DAGNode, context, index=None):
                     index=index,
                 ).set_display_name(node.name)
 
+                resource = KubeflowPipelines._get_resource_requirements(node)
+                if "memory" in resource:
+                    visited[node.name].set_memory_request(resource["memory"])
+                if "memory_limit" in resource:
+                    visited[node.name].set_memory_limit(resource["memory_limit"])
+                if "cpu" in resource:
+                    visited[node.name].set_cpu_request(resource["cpu"])
+                if "cpu_limit" in resource:
+                    visited[node.name].set_cpu_limit(resource["cpu_limit"])
+                if "gpu" in resource:
+                    # TODO(yunw)(AIP-2048): Support mixture of GPU from different vendors.
+                    visited[node.name].set_gpu_limit(
+                        resource["gpu"], vendor=resource["gpu_vendor"]
+                    )
```

Comment:
This block of code could be a nice function.

`_set_k8s_resources(node, container_op)`